lib/go-atscfg/remapdotconfig_test.go

```diff
@@ -2085,15 +2085,15 @@ func TestMakeRemapDotConfigEdgeHostRegexReplacement(t *testing.T) {
 
 	txtLines := strings.Split(txt, "\n")
 
-	if len(txtLines) != 3 {
+	if len(txtLines) != 4 {
 		t.Fatalf("expected 3 remaps from HTTP_AND_HTTPS DS, actual: '%v' count %v", txt, len(txtLines))
```

Comment:
> "expected <del>3</del><ins>4</ins> remaps from ..."

Reply:
But currently if you have 3 remaps but no blank line as output then this will print out "expected 3 remaps ... actual:" and then three remaps and a count of 3, so what the problem is remains a mystery to anyone reading it. The text doesn't reflect what's actually expected.

Reply:
Ok, that should actually say 2 remaps, not 3. I fixed. 

But I'm still having trouble understanding the confusion here.

> what the problem is remains a mystery to anyone reading it. The text doesn't reflect what's actually expected.

I'm not understanding that. The text explains exactly what the problem is, what's expected, and what the actual result was.

So, I changed the number there to "42" to force the test to fail. If the test fails, it outputs:

```
=== RUN   TestMakeRemapDotConfigEdgeHostRegexReplacement
    remapdotconfig_test.go:2089: expected 3 remaps from HTTP_AND_HTTPS DS, actual: '# myHeaderComment
        
        map     http://server0.mypattern.cdndomain.example:12080/     origin.example.test/ @plugin=header_rewrite.so @pparam=dscp/set_dscp_0.config @plugin=url_sig.so @p
param=url_sig_mydsname.config @plugin=cachekey.so @pparam=--separator= @pparam=--remove-all-params=true @pparam=--remove-path=true @pparam=--capture-prefix-uri=/^([^?]*)
/$1/ @plugin=regex_remap.so @pparam=regex_remap_mydsname.config @plugin=cache_range_requests.so  myremaptext
        map     https://server0.mypattern.cdndomain.example:12345/     origin.example.test/ @plugin=header_rewrite.so @pparam=dscp/set_dscp_0.config @plugin=url_sig.so @
pparam=url_sig_mydsname.config @plugin=cachekey.so @pparam=--separator= @pparam=--remove-all-params=true @pparam=--remove-path=true @pparam=--capture-prefix-uri=/^([^?]*
)/$1/ @plugin=regex_remap.so @pparam=regex_remap_mydsname.config @plugin=cache_range_requests.so  myremaptext' count 4
--- FAIL: TestMakeRemapDotConfigEdgeHostRegexReplacement (0.00s)
```

It's immediately obvious that there are 2 remaps there (or presumably not 2, if it's legitimately failing). The failure message explains that 2 are expected because it's a single HTTP_AND_HTTPS DS.

The count is the line count, which should be obvious from the test code. I guess someone could potentially be confused if they miss that. I guess we could change that text to "line count." But that still feels like a bit of a stretch. What's the confusion here?

Reply:
> What's the confusion here?

I find it confusing when a test tells me "This value should have been X, but it was X" - that reads like the expectation was satisfied, which leads me to believe there's something wrong with the test and not my code.

If I make a change to `MakeRemapDotConfig` such that for the same configuration as given on line 2076 of `remapdotconfig_test.go` it outputs
```
map     http://server0.mypattern.cdndomain.example:12080/     origin.example.test/ @plugin=header_rewrite.so @pparam=dscp/set_dscp_0.config @plugin=url_sig.so @p
param=url_sig_mydsname.config @plugin=cachekey.so @pparam=--separator= @pparam=--remove-all-params=true @pparam=--remove-path=true @pparam=--capture-prefix-uri=/^([^?]*)
/$1/ @plugin=regex_remap.so @pparam=regex_remap_mydsname.config @plugin=cache_range_requests.so  myremaptext
map     https://server0.mypattern.cdndomain.example:12345/     origin.example.test/ @plugin=header_rewrite.so @pparam=dscp/set_dscp_0.config @plugin=url_sig.so @
pparam=url_sig_mydsname.config @plugin=cachekey.so @pparam=--separator= @pparam=--remove-all-params=true @pparam=--remove-path=true @pparam=--capture-prefix-uri=/^([^?]*
)/$1/ @plugin=regex_remap.so @pparam=regex_remap_mydsname.config @plugin=cache_range_requests.so  myremaptext
```
without a header comment at all, the resulting error message will read
```
=== RUN   TestMakeRemapDotConfigEdgeHostRegexReplacement
    remapdotconfig_test.go:2089: expected 2 remaps from HTTP_AND_HTTPS DS, actual: 'map     http://server0.mypattern.cdndomain.example:12080/     origin.example.test/ @plugin=header_rewrite.so @pparam=dscp/set_dscp_0.config @plugin=url_sig.so @p
param=url_sig_mydsname.config @plugin=cachekey.so @pparam=--separator= @pparam=--remove-all-params=true @pparam=--remove-path=true @pparam=--capture-prefix-uri=/^([^?]*)
/$1/ @plugin=regex_remap.so @pparam=regex_remap_mydsname.config @plugin=cache_range_requests.so  myremaptext
        map     https://server0.mypattern.cdndomain.example:12345/     origin.example.test/ @plugin=header_rewrite.so @pparam=dscp/set_dscp_0.config @plugin=url_sig.so @
pparam=url_sig_mydsname.config @plugin=cachekey.so @pparam=--separator= @pparam=--remove-all-params=true @pparam=--remove-path=true @pparam=--capture-prefix-uri=/^([^?]*
)/$1/ @plugin=regex_remap.so @pparam=regex_remap_mydsname.config @plugin=cache_range_requests.so  myremaptext' line count 2
--- FAIL: TestMakeRemapDotConfigEdgeHostRegexReplacement (0.00s)
```
So it
- expected 2 remaps
- got 2 remaps
- on 2 lines

and from that I can't tell what it was supposed to output or why the shown output is wrong. This testing error message does not contain any information about why the test failed.

> ... should be obvious from the test code. I guess someone could potentially be confused if they miss that.

If I have to look at the test code to understand what expectation failed, then why bother writing an error message at all instead of just outputting the erroneous value? Which, IMO is a totally valid thing to do and I've done it many times in Angular unit tests. But outputting a misleading error message is worse than none at all.

Reply:
So the potential confusion is that it could output 2 remap lines and fail for other reasons? Would changing the error message to `t.Fatalf("expected a comment header line, a blank line, and 2 remaps from HTTP_AND_HTTPS DS, actual:` address the concern?

Reply:
Changed